src/App.tsx

```diff
@@ -1,222 +1,302 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+
+import React, {
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+import {
+  getTodos,
+  postTodos,
+  deleteTodos,
+  patchTodos,
+} from './api/todos';
+
+import { Filter, Filters } from './components/React/Filter';
 import { AuthContext } from './components/Auth/AuthContext';
+import { NewTodo } from './components/React/NewTodo';
+import { TodoList } from './components/React/TodoList';
+import { Todo } from './types/Todo';
+
+enum Errors {
+  Server = 'Unable to load data from the Server',
+  Post = 'Unable to add the Todo',
+  Add = 'Title can\'t be empty',
+  Delete = 'Unable to delete a todo',
+  Update = 'Unable to update a todo',
+}
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<Filters>(Filters.All);
+  const [error, setError] = useState('');
+  const [input, setInput] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [todoOnload, setTodoOnLoad] = useState<Todo | null>(null);
+  const [todoIdsOnLoad, setTodoIdsOnLoad] = useState<number[]>([]);
+  const [toggle, setToggle] = useState(false);
+
+  const putTodoOnLoad = (todoId: number) => {
+    setTodoIdsOnLoad([...todoIdsOnLoad, todoId]);
+  };
+
+  const putAllTodosOnLoad = (selectedTodos: Todo[]) => {
+    const selectedTodoIds = selectedTodos.map(todo => todo.id);
+
+    setTodoIdsOnLoad(selectedTodoIds);
+  };
+
+  const fetch = () => {
+    if (user) {
+      getTodos(user.id)
+        .then(todosFromServer => setTodos(todosFromServer))
+        .catch(() => setError(Errors.Server))
+        .finally(() => setTodoIdsOnLoad([]));
+    }
+  };
+
+  const deleteData = (id: number) => {
+    deleteTodos(id)
+      .then(() => fetch())
+      .catch(() => setError(Errors.Delete));
+  };
+
+  const patch = async (
+    id: number,
+    data: boolean | string,
+    toggleMode = false,
+  ) => {
+    let updatedTodos: Todo[];
+
+    if (todoIdsOnLoad.length) {
+      return;
+    }
+
+    try {
+      putTodoOnLoad(id);
+
+      await patchTodos(
+        id,
+        typeof data === 'string'
+          ? { title: data }
+          : { completed: !data },
```

Comment:
better to create a separate variable instead of ternary operator in function parameters